tools/generate-helpers/pg-table-bindings/store.go.tpl

```diff
@@ -258,12 +258,18 @@ func {{ template "copyFunctionName" $schema }}(ctx context.Context, s pgSearch.D
     if len(objs) == 0 {
         return nil
     }
-    batchSize := min(len(objs), pgSearch.MaxBatchSize)
-    inputRows := make([][]interface{}, 0, batchSize)
     {{if not $schema.Parent }}
-    // This is a copy so first we must delete the rows and re-add them
-    // Which is essentially the desired behaviour of an upsert.
-    deletes := make([]string, 0, batchSize)
+    {
+        // CopyFrom does not upsert, so delete existing rows first to achieve upsert behavior.
+        // Parent deletion cascades to children, so only the top-level parent needs deletion.
+        deletes := make([]string, 0, len(objs))
```

Comment:
We should still batch the deletes.  While unlikely it is possible to get more than 65K at once which would fail.  It has also generally been faster to delete in smaller chunks than huge chunks.

Reply:
Good point and we already handle that on `DeleteMany` level:

https://github.com/stackrox/stackrox/blob/5fbb8fd2a569b79a38a8e3d885890b2402c6a599/pkg/search/postgres/store.go#L27
https://github.com/stackrox/stackrox/blob/5fbb8fd2a569b79a38a8e3d885890b2402c6a599/pkg/search/postgres/store.go#L365

Reply:
fantastic.